src/main/java/com/puppycrawl/tools/checkstyle/checks/TrailingCommentCheck.java

```diff
@@ -289,18 +286,9 @@ private void checkBlockComment(DetailAST ast) {
      * @return true if the comment if legal.
      */
     private boolean isLegalBlockComment(DetailAST ast, String comment) {
-        final boolean legal;
-
-        // multi-line comment can not be legal
-        if (legalComment == null
-                || !TokenUtil.areOnSameLine(ast.getFirstChild(), ast.getLastChild())) {
-            legal = false;
-        }
-        else {
-            final String commentText = comment.trim();
-            legal = legalComment.matcher(commentText).find();
-        }
-        return legal;
+        return legalComment != null
+                && TokenUtil.areOnSameLine(ast.getFirstChild(), ast.getLastChild())
```

Comment:
Please do me favor and move this line out of this method.
Method should care only about content of message, as defined in name and similar method for single line does

Reply:
Check logic dictates that any trailing mutliline comment cannot be legal, tokens being on same line makes sense in this case to me.

Reply:
Please move it out, as separate commit.
It  might be good to name methods `isLegalCommentContent` as it is based on legal comment field, and this field care about content only.

Reply:
Done in separate commit, other refactoring was necessary or was obvious improvement.